src/any_llm/providers/bedrock/bedrock.py

```
@@ -115,10 +119,51 @@ def _convert_list_models_response(response: Any) -> Sequence[Model]:
         if self._custom_client is not None:
             self.client = self._custom_client
-        else:
-            self.client = boto3.client("bedrock-runtime", endpoint_url=api_base, **kwargs)
+            self._boto_session = None
+            return
+        self._boto_session = self._build_boto_session(api_key)
+        self.client = self._boto_session.client(
+            "bedrock-runtime", endpoint_url=api_base, **self._bedrock_client_kwargs(kwargs)
+        )
+
+    def _build_boto_session(self, api_key: str | None) -> Any:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_build_boto_session`

(ANN401)